src/App.tsx

```diff
@@ -1,24 +1,147 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/Errors';
+import * as TodoServices from './api/todos';
+import { FilterType } from './types/FilterType';
+import { getPreparedTodos } from './utils/filterFunction';
+import { TodoList } from './Components/TodoList';
+import { TodoFooter } from './Components/TodoFooter';
+import { TodoHeader } from './Components/TodoHeader';
+import { ErrorNotification } from './Components/ErrorsNotification';
 
-const USER_ID = 0;
+const USER_ID = 11128;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorType>(ErrorType.none);
+  const [filterType, setFilterType] = useState(FilterType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [loadingId, setLoadingId] = useState<number[]>([]);
+
+  const visibleTodos = useMemo(
+    () => getPreparedTodos(todos, filterType),
+    [todos, filterType],
+  );
+
+  const complitedTodods = useMemo(
+    () => todos.every(todo => todo.completed), [todos],
+  );
+
+  useEffect(() => {
+    TodoServices.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setError(ErrorType.getData);
+      });
+  }, []);
+
+  const handleAddTodo = ({ title, completed, userId }: Todo) => {
+    setLoading(true);
+
+    return TodoServices.createTodo({ title, completed, userId })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+      })
+      .catch((newError) => {
+        throw newError;
+      })
+      .finally(() => setLoading(false));
+  };
+
+  const handleUpdateTodo = (todoId: number, data: Partial<Todo>) => {
+    setLoadingId(curId => [todoId, ...curId]);
+
+    return TodoServices.updateTodo(todoId, data)
+      .then(todo => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(t => t.id === todoId);
+
+          newTodos.splice(index, 1, todo);
+
+          return newTodos;
+        });
+      })
+      .catch((newError) => {
+        setError(ErrorType.updateTodo);
+        throw newError;
```

Comment:
```suggestion
```

Reply:
there are i need 'throw newError;' because i cathe it in TodoItem component.